dev-tools/gen-target-info/src/target_specs.rs

```diff
@@ -32,3 +35,48 @@ pub struct RustcTargetSpecs(
     /// First field in the tuple is the rustc target
     pub BTreeMap<String, TargetSpec>,
 );
+
+/// Potentially useful values from:
+/// <https://doc.rust-lang.org/reference/conditional-compilation.html>
+/// That are not directly / easily exposed in `TargetSpec`.
+#[derive(Debug, Default)]
+pub struct Cfgs {
+    pub target_features: Vec<String>,
+    pub target_families: Vec<String>,
+    pub target_endian: String,
+    pub target_atomics: Vec<String>,
+    pub target_thread_local: bool,
+}
+
+impl Cfgs {
+    pub fn parse(cfgs: &[String]) -> Self {
+        let mut target_features = vec![];
+        let mut target_families = vec![];
+        let mut target_endian = None;
+        let mut target_atomics = vec![];
+        let mut target_thread_local = false;
+        for cfg in cfgs {
+            let (name, value) = cfg
+                .split_once('=')
+                .map(|(n, v)| (n.trim(), Some(v.trim().trim_matches('"'))))
+                .unwrap_or((cfg.trim(), None));
+
+            match (name, value) {
+                ("target_feature", Some(value)) => target_features.push(value.to_string()),
+                ("target_family", Some(value)) => target_families.push(value.to_string()),
+                ("target_endian", Some(value)) => target_endian = Some(value.to_string()),
```

Comment:
Nit:

Shall we have a check for duplicate value, just in case rustc gives us some invalid output?

Reply:
Good idea